assemblyline_v4_service/updater/updater.py

```diff
@@ -332,7 +332,11 @@ def do_local_update(self) -> None:
             self.log.info(f"Connecting to Assemblyline API: {UI_SERVER}")
             al_client = get_client(UI_SERVER, apikey=(username, api_key), verify=self.verify, datastore=self.datastore)
 
-            _, time_keeper = tempfile.mkstemp(prefix="time_keeper_", dir=UPDATER_DIR)
+            # Create a temporary file for the time keeper
+            time_keeper = tempfile.NamedTemporaryFile(prefix="time_keeper_", dir=UPDATER_DIR, delete=False)
+            time_keeper.close()
+            time_keeper = time_keeper.name
```

Comment:
Pedantic thought but I wonder if there is any possibility of an interrupt or exception happening between file creation and the close call that a with block would avoid. I can't think of any that wouldn't also kill the process (CTRL-C), which would free the file handle anyway. So it's probably irrelevant.